ONSdigital/dp-design-system
# Feature/refactor search js

### What

[Refactored search JS](https://jira.ons.gov.uk/browse/DIS-1125) and [Fix js on /previousReleases](https://jira.ons.gov.uk/browse/DIS-1780)
AKA
- Broke files into subdirectory `search` with files:
  - `_helpers.js` i.e helper functions for search
  - `_filter_handlers.js` i.e. functions that handler various filter components
  - `_timeseries_handler` i.e. file to handle the timeseries basket
  - `search.js` i.e. primary search file where hooks are instantiated 
- Removed redundant code i.e. markup hooks changed/removed or code no longer used
- Fixed pagination bug on `/previousreleases`

Hopefully 🤞 the file naming convention makes sense to what the functions within the files do. I've deliberately made them child files of search as the functions are quite specific to search

### How to review

Sense check
Run locally
- port forward to sandbox api router
- make debug on the `dp-frontend-search-controller` with the following feature toggles set to `true`:
  - EnableCensusTopicFilterOption
  - EnableCensusPopulationTypesFilterOption (supporting js code removed)
  - EnableCensusDimensionsFilterOption (supporting js code removed)
  - EnableAggregationPages
  - EnableTopicAggregationPages
- pull this branch and run `nvm use && npm run dev`
- checkout 👀 various search endpoints and check the interactions; there should be no js errors for search
- suggested endpoints:
  - http://localhost:25000/alladhocs
  - http://localhost:25000/economy/grossdomesticproductgdp/bulletins/gdpfirstquarterlyestimateuk/previousreleases
  - http://localhost:25000/timeseriestool
  - http://localhost:25000/search?q=census

### Who can review

Frontend dev
